### ISSUE TITLE
Unable to remove root verification on Chrome OS v70

### ISSUE THREAD

**Issue by Giinx** [nolirium/aroc#10]
I would like some help because I am unable to remove root verification on this version, when I run the command: **sudo /usr/share/vboot/bin/make_dev_ssd.sh --remove_rootfs_verification --partitions $(( $(rootdev -s | sed -r 's/.*(.)$/\1/') - 1))** everythings appears to be okay but when I restart my chromebook and try to run RootandSEpatch.sh I get the error "_Error!
Unable to modify system!
In order to modify system files, the Chrome OS system partition needs to have been mounted writeable (i.e. rootfs verification disabled)._"

I would like some help with this please.

**Comment by nolirium** [comment]
Yes, the rootfs verification check in the script got broken due to a change in Chrome OS. The RootandSEpatch.sh script has just been updated with a fix - if you try it again now, it should work this time.

**Comment by Giinx** [comment]
chronos@localhost / $ curl -Ls https://raw.githubusercontent.com/nolirium/aroc/onescript/RootandSEpatch.sh | sudo sh
sh: 118: Syntax error: "fi" unexpected (expecting "then")

**Comment by nolirium** [comment]
Should be fixed now.

**Comment by Giinx** [comment]
> Should be fixed now.

cheers, so I did that and it still didnt work the chromebook running chromeos v70 is still unrooted for some reason even after doing this.

**Comment by nolirium** [comment]
Did you happen to notice if there were any error messages in the script's output? 

Are Android apps working now (but no root), or not loading at all, or ...?

**Comment by Giinx** [comment]
> Did you happen to notice if there were any error messages in the script's output?
> 
> Are Android apps working now (but no root), or not loading at all, or ...?

normal android apps that require no root work fine, everything is still working just fine as normal but everything that requires root and any root checks come up as device unrooted still, and there were no error messages but I can do it again now and screenshot the logs for you

**Comment by Giinx** [comment]
Got an error 
ERROR: No running Android system found. Unable to patch sepolicy.

**Comment by nolirium** [comment]
Yeah, that'll happen if for example you try to run the script twice over without rebooting between times. It's expected, in that case. 

Try rebooting again, and if still working and unrooted, can you paste the following into the Chrome OS shell and see if anything comes up:
```
sudo su -
printf "ps | grep daemon" | android-sh
```

**Comment by Giinx** [comment]
I ran the script the first time then I rebooted and it wasnt rooted so I went in shell and tried again so I can check if there were any errors for you and thats the error that I got :) But I did reboot right away the first time I ran the script

**Comment by Giinx** [comment]
> Yeah, that'll happen if for example you try to run the script twice over without rebooting between times. It's expected, in that case.
> 
> Try rebooting again, and if still working and unrooted, can you paste the following into the Chrome OS shell and see if anything comes up:
> 
> ```
> sudo su -
> printf "ps | grep daemon" | android-sh
> ```

infinite loading on apps now wont seem to work anymore

**Comment by nolirium** [comment]
OK well give it a while (it takes some time after the first reboot) and if apps are still not loading the command to restore the unrooted Android is

```
sudo mv /opt/google/containers/android/system.raw.img.bk /opt/google/containers/android/system.raw.img
```

If you paste that into the Chrome OS shell and then reboot, you should have the normal unrooted Android system. 

I may not be able to respond back until tomorrow now, but if you do end up unrooting per the command above, and then running the script again, it might be useful if you could save the script's output to text (or screengrab), so we could check if there were any errors running the script.

**Comment by Giinx** [comment]
It's starting to get really annoying now it doesnt work at all anymore, I used to command to restore the unrooted android container then rebooted and tried to run RootandSEpatch.sh but it just got all messed up I have attached the pictures.
![screenshot 2018-11-18 at 4 01 18 am](https://user-images.githubusercontent.com/45128077/48667778-aec00e80-eae6-11e8-9741-1a814dd32c34.png)
![screenshot 2018-11-18 at 4 02 09 am](https://user-images.githubusercontent.com/45128077/48667784-c4cdcf00-eae6-11e8-8125-e512338c68a7.png)

**Comment by nolirium** [comment]
I know it's a hassle to keep rebooting and logging in, but can you just try rebooting once more at this point and then running the script exactly the same way.

**Comment by Giinx** [comment]
Should I try to run the unroot script then the root script once more?

**Comment by Giinx** [comment]
I can go on root@localhost ~ but SuperSU and Root Checker still say my device isnt rooted and I cant get the rooted apps working

**Comment by nolirium** [comment]
But everything else (file copying etc) looks like it worked OK. Hmm...

If you do 

`sudo android-sh`

does it get you into an Android shell like it's supposed to? (you would see the prompt change from "chronos@localhost: " to "something_cheets:")

Or does it just give an error message?

**Comment by Giinx** [comment]
it takes me to edgar_sheets:/

**Comment by nolirium** [comment]
OK that's good. Now, at the edgar_cheets prompt, can you put in
`ps | grep daemon`
and tell me what, if anything, shows up?

**Comment by Giinx** [comment]
![screenshot 2018-11-18 at 2 57 31 pm](https://user-images.githubusercontent.com/45128077/48672732-4e5eba80-eb42-11e8-8de3-69a63ab9cf41.png)

**Comment by nolirium** [comment]
So daemonsu isn't running. Can you try to start it (see if it is present)? Again in the edgar_cheets prompt, do

`daemonsu --auto-daemon
`

**Comment by Giinx** [comment]
![screenshot 2018-11-18 at 4 02 58 pm](https://user-images.githubusercontent.com/45128077/48673579-70a90600-eb4b-11e8-8d30-cebfae545ea0.png)

**Comment by Giinx** [comment]
![screenshot 2018-11-18 at 4 04 03 pm](https://user-images.githubusercontent.com/45128077/48673591-96cea600-eb4b-11e8-83d7-8100709ff13f.png)

**Comment by nolirium** [comment]
Try a root app now - see if it asks for permission.

**Comment by nolirium** [comment]
Something's not right, clearly. Maybe we should try to patch the SE Linux policy file manually, to rule out SE Linux being the problem (since [one of your screenshots](https://user-images.githubusercontent.com/45128077/48667924-6f93bc80-eaea-11e8-8fcd-5aa7687ccf42.png) showed the rooting script erroring out at the point, right near the end where it's supposed to patch the sepolicy file).

Here's a script which is just the SE Linux policy patching part. (you can run it in the Chrome OS shell)

```
curl -Ls https://raw.githubusercontent.com/nolirium/aroc/master/Test/sepatchtest.sh | sudo sh
```

**Comment by Giinx** [comment]
![screenshot 2018-11-18 at 5 13 33 pm](https://user-images.githubusercontent.com/45128077/48674462-4fe5ae00-eb55-11e8-9486-af3297b38e66.png)

**Comment by nolirium** [comment]
Maybe you could double check the contents of /system/xbin in Android:

```
sudo android-sh
ls -alt /system/xbin
```

Also in the Android shell (the edgar_cheets prompt), could you do

```su --version```

(It should say "2.82:SUPERSU")

And also at the edgar_cheets prompt, if you just do

```su```

and see if you get an error there or if the edgar_cheets prompt changes at all (if any number shows up).

**Comment by Giinx** [comment]
Would turning this on make a difference? Just wondering.
![screenshot 2018-11-18 at 5 27 39 pm](https://user-images.githubusercontent.com/45128077/48674665-4fe6ad80-eb57-11e8-9a5d-53108fbceffe.png)

**Comment by Giinx** [comment]
update on your last comment btw
![screenshot 2018-11-18 at 5 29 52 pm](https://user-images.githubusercontent.com/45128077/48674686-9805d000-eb57-11e8-9a0d-f6244ca69336.png)

**Comment by nolirium** [comment]
> Would turning this on make a difference? Just wondering.
> ![screenshot 2018-11-18 at 5 27 39 pm](https://user-images.githubusercontent.com/45128077/48674665-4fe6ad80-eb57-11e8-9a5d-53108fbceffe.png)

It shouldn't do any harm, but unless something has changed very recently, it shouldn't affect Android either way.

**Comment by Giinx** [comment]
would turning that on fix our issue or what do you recommend?

**Comment by nolirium** [comment]
> would turning that on fix our issue or what do you recommend?

At the moment we don't know what the exact cause of the issue is, but turning on Linux apps is unlikely to affect it. Feel free to try it though, it shouldn't do any harm...


>update on your last comment btw
>![screenshot 2018-11-18 at 5 29 52 pm](https://user-images.githubusercontent.com/45128077/48674686-9805d000-eb57-11e8-9a0d-f6244ca69336.png)

So, it looks like the files for SuperSU are all present and correct. The "1|edgar_cheets" that appears at the front of your last command indicates an error, but at least Android is booting and the files appear to be in the right place.

I'm going to go look through some of the comments from my blog, as I remember some people had similar problems a while ago, with various causes (and resolutions). I'm thinking perhaps the modification to init.rc is at fault, but perhaps it's something else entirely.

One last easy thing that comes to mind to maybe try is to disable enforcing SELinux completely (then reboot). 

`sed -i 's/SELINUX=enforcing/SELINUX=permissive/g' /etc/selinux/config`

**Comment by Giinx** [comment]
![screenshot 2018-11-18 at 5 57 41 pm](https://user-images.githubusercontent.com/45128077/48674979-773f7980-eb5b-11e8-8bef-875d87742242.png)
sed: couldn't open temporary file /etc/selinux/sedczIHPC: Permission denied

**Comment by nolirium** [comment]
Oh, forgot to mention that you'll need to `sudo su` beforehand for that one to work

**Comment by Giinx** [comment]
sweet dude it's working now we got it done!! 👍 
![screenshot 2018-11-18 at 6 28 33 pm](https://user-images.githubusercontent.com/45128077/48675288-cf787a80-eb5f-11e8-8e03-6f8f30cc1444.png)
after running that command I rebooted got back on opened supersu updated the packages rebooted once again and now it's working fine it seems like I'll keep u updated, let me know if u want me to test out some more stuff tho

**Comment by nolirium** [comment]
Good stuff! Glad we could get it sorted. 

I'll still have to try and work out out why it wasn't working initially. Possibly the script still needs tweaking somewhat.

For the moment though, it might be helpful towards figuring out the initial cause if you could do one more thing now, which is to redo the SE Linux patching command mini script in the Chrome OS shell, and see if it succeeds this time :

```
curl -Ls https://raw.githubusercontent.com/nolirium/aroc/master/Test/sepatchtest.sh | sudo sh
```

(If it tells you to reboot, just ignore that).

**Comment by Giinx** [comment]
OK just an update before I do what you told me to do, after I took the screenshot that I showed u I rebooted and now that I'm back on it it seems like apps aren't opening anymore it's just an infinite loading loop

**Comment by Giinx** [comment]
Just ran the script that u sent me btw
![screenshot 2018-11-18 at 7 38 01 pm](https://user-images.githubusercontent.com/45128077/48676032-7f9eb100-eb69-11e8-9c43-b544a34e9771.png)

**Comment by nolirium** [comment]
> Just ran the script that u sent me btw
> ![screenshot 2018-11-18 at 7 38 01 pm](https://user-images.githubusercontent.com/45128077/48676032-7f9eb100-eb69-11e8-9c43-b544a34e9771.png)

Well, that appears to have worked, which is something.

Did the previous command stick? If you type 

```
sudo su
grep SELINUX= /etc/selinux/config 
``` 

into the Chrome OS shell, does it show 'Permissive' or 'Enforcing'?

How are you rebooting, by the way? I mean, are you just typing `reboot`  into the shell, or using the power button, or something else? (I only ask as I remember one person who was pressing Esc+Refresh+Power instead of a regular reboot every time, which was messing things up in strange ways).

**Comment by Giinx** [comment]
I reboot by pressing the power button + the spin button on my keyboard and I did what u told me ran the command it says permissive

**Comment by nolirium** [comment]
When you say the spin button, do you mean the 'refresh' key (F3)? If so... Don't do that. 

Just use the power button (or type `reboot` in a root shell). 

Pressing the power and refresh keys both together can potentially do a [hard reset](https://support.google.com/chromebook/answer/3227606?hl=en-GB) if you keep them held down for too long, which can really mess things up if you've just made some changes, as the system will completely 'forget' the changes.

**Comment by Giinx** [comment]
I didn't hold it down for too long though I only pressed them twice together once and it restarts, what should I do now though? because when I do reboot in a root shell it says command not found

**Comment by Giinx** [comment]
after checking im pretty sure power button + refresh key is fine btw Im only clicking it for a millisecond

**Comment by nolirium** [comment]
> after checking im pretty sure power button + refresh key is fine btw Im only clicking it for a millisecond

Yeah, it was probably fine then - you would have had to have been holding them down for a moment to hard reset, and IIRC the screen normally flashes white briefly, so you'd have probably noticed. But there's no need to press the refresh button anyway, so...

>  when I do reboot in a root shell it says command not found

Were you doing it in a root shell, or as chronos@localhost, though? 

Just to clarify, within the Chrome OS shell terminal:

This is the Chrome OS regular user shell prompt:
chronos@localhost / $ 

This is the Chrome OS root user shell prompt:
localhost / # 

(If you try to run a command and you get a permission error, or if it says command not found, sometimes  it's because you're in the shell as a regular user (in this case, chronos). If so, you can change to root with `sudo su`).

Apologies if this is all stating the obvious.

Anyway... Are you still in the same situation? Have you rebooted (either way) since you ran the last script?

**Comment by Giinx** [comment]
yup still having the same issue guess its time to reset and start over.. god.

**Comment by Giinx** [comment]
did a powerwash and now I cant get playstore working again it's getting really annoying, even though I ran the command to restore the unrooted android container

**Comment by nolirium** [comment]
> did a powerwash and now I cant get playstore working again it's getting really annoying, even though I ran the command to restore the unrooted android container

So Android isn't loading at all after a powerwash? What is happening, just the loading graphic constantly spinning?
You might try taking a look at logcat if you can get into android-sh from the Chrome OS root shell 

```printf logcat | android-sh``` 

just to see if it is looping or frozen (normally it's constantly updating).

You could also try double checking that the restore command worked (by listing the files in /opt/google/containers/android) 

```ls -alt /opt/google/containers/android/```

The nuclear option would be to either switch release channel, or do a restore with the Recovery Utility. Recovery should pretty much always work, as it'll download the whole OS again, and the release channel change usually works (but in certain situations can fail). A powerwash only deletes user files, mainly.

**Comment by Giinx** [comment]
google services just wont work anymore for some reason I attached 2 pictures of how the playstore is behaving, if I close it and try to reopen it, it simply doesnt open and goes into infinite loop
![screenshot 2018-11-18 at 10 20 05 pm](https://user-images.githubusercontent.com/45128077/48677622-39a11780-eb80-11e8-99ef-daef6205241a.png)
![screenshot 2018-11-18 at 10 20 00 pm](https://user-images.githubusercontent.com/45128077/48677623-3a39ae00-eb80-11e8-9111-ee5bfde61d26.png)

**Comment by Giinx** [comment]
Should I try going into the dev channel?

**Comment by nolirium** [comment]
In your first image there, you are trying to run the command `printf logcat | android-sh` as chronos, but it needs to be done in a root Chrome OS shell (do `sudo su` first).

> Should I try going into the dev channel?

That should work, but first you could maybe just try toggling between Enforcing and Permissive SELinux again, then rebooting.

Again in a root Chrome OS shell, If you do `grep SELINUX= /etc/selinux/config` again, if it still shows as 'Permissive', you can change it back to 'Enforcing' (its original state) with:

`sed -i 's/SELINUX=permissive/SELINUX=enforcing/g' /etc/selinux/config
`

**Comment by Giinx** [comment]
alright give me a bit I'll get it sorted now I'll go into dev channel then back to stable that should work and then when google services work again I'll go through all the steps again to root it okay? I'll keep you updated

**Comment by nolirium** [comment]
> alright give me a bit I'll get it sorted now I'll go into dev channel then back to stable that should work and then when google services work again I'll go through all the steps again to root it okay? I'll keep you updated.

Seems like a good plan. Although, without wanting to add a whole other layer of potential confusion, I should also just mention in case you weren't aware that Chrome OS uses an A/B partition system for updates, i.e. lets say you start off on partition 3, the next update sets a kernel flag so you will be using partition 5 after rebooting, and after the next update on partition 3, and so on. 

They do this so they can apply updates smoothly, and in normal use for the most part it works fine, but occasionally it can lead to messed up/confusing situations when removing root verification/modifying system files & rebooting.  Going stable>dev>stable should be fine, but I thought I'd mention the A/B partition system just in case.

**Comment by Giinx** [comment]
ok man its all good now I went stable>dev>stable and then ran the rootandsepatch script rebooted and it just worked this time didnt have to do anything else, and also about the A/B partition how will this affect me in the future? and is there anyway I can just make it go back to normal or something

**Comment by Giinx** [comment]
I'm too scared to restart or turn off my chromebook now btw. lol :D

**Comment by nolirium** [comment]
> ok man its all good now I went stable>dev>stable and then ran the rootandsepatch script rebooted and it just worked this time didnt have to do anything else, 

Good. That's how it's supposed to work. 

>and also about the A/B partition how will this affect me in the future? and is there anyway I can just make it go back to normal or something

Oh, it is normal, it's how they've designed the update system. It most likely won't affect you ever unless if, for instance, you were running the rooting script while you had an update waiting, or maybe you are hacking around on the OS, changing other stuff around, then e.g. you change channel a few times real quickly. Even in that case, all that would likely happen would you'd need to re-do the thing you changed (it's just that if it does happen, it can be completely confusing at the time).

> I'm too scared to restart or turn off my chromebook now btw. lol :D

haha, yeah I can understand that after all of this. Although if you reboot it once, and everything still works, then at least you'll know it'll be OK from there on. Normally once it works, that's it, everything's fine (until the next major Chrome OS update, anyway).

**Comment by Giinx** [comment]
> > ok man its all good now I went stable>dev>stable and then ran the rootandsepatch script rebooted and it just worked this time didnt have to do anything else,
> 
> Good. That's how it's supposed to work.
> 
> > and also about the A/B partition how will this affect me in the future? and is there anyway I can just make it go back to normal or something
> 
> Oh, it is normal, it's how they've designed the update system. It most likely won't affect you ever unless if, for instance, you were running the rooting script while you had an update waiting, or maybe you are hacking around on the OS, changing other stuff around, then e.g. you change channel a few times real quickly. Even in that case, all that would likely happen would you'd need to re-do the thing you changed (it's just that if it does happen, it can be completely confusing at the time).
> 
> > I'm too scared to restart or turn off my chromebook now btw. lol :D
> 
> haha, yeah I can understand that after all of this. Although if you reboot it once, and everything still works, then at least you'll know it'll be OK from there on. Normally once it works, that's it, everything's fine (until the next major Chrome OS update, anyway).

what if a SuperSU update happens and u reboot then apps are stuck on infinite loop again? man I'm cursed lol

**Comment by gyit918** [comment]
when i do sudo android-sh
it gives me this sudo 
:/ #
